### PR TITLE
[TS-12]: build validation for create account screen

### PR DESCRIPTION
**Pull request by AnoshaKhan**
#### Discription:
1: Next button disabled until user fill all the information.
2: Email has valid format such as "johndoe@gmail.com".
3: Password has been secured  and will not be visible all the time.
4: Password has a special character and a number and no emoji.
5: User is restricted to enter more than 6 letters in password.
6: Keyboard will dismissed when the user will tap on the screen.
7: The screen is scrollable so that keyboard
 will not cover the button and textfield.

#### How to test it ?
 Go to info.plist and change the storyboard name to CreateAccountScreen

### CODE REVIEW

File: ShopGo/CreateScreenViewController.swift [hunk 1]
```diff
@@ -17,18 +17,44 @@ class CreateScreenViewController: UIViewController {
     let emailImage = UIImage(named: "mail")
     let passwordImage = UIImage(named: "password")
     let phoneNoImage = UIImage(named: "phone")
-    let userNameImage = UIImage(named: "user")
-    
+    let userNameImage = UIImage(named: "person")
+   
     override func viewDidLoad() {
         super.viewDidLoad()
+
```
Comment (zuhair5) [inline ShopGo/CreateScreenViewController.swift:24]: remove space

File: ShopGo/CreateScreenViewController.swift [hunk 2]
```diff
@@ -17,18 +17,44 @@ class CreateScreenViewController: UIViewController {
...
+        nextButton.isEnabled = false
+        nextButton.alpha = 0.5
+
```
Comment (zuhair5) [inline ShopGo/CreateScreenViewController.swift:27]: remove space

File: ShopGo/CreateScreenViewController.swift [hunk 3]
```diff
@@ -39,10 +65,74 @@ class CreateScreenViewController: UIViewController {
...
     }
     
+    // textField image setup
     func addLeftImage(textField: UITextField , image: UIImage) {
         let leftImageView = UIImageView(frame: CGRect(x: 0.0, y: 0.0, width: image.size.width, height: image.size.height))
         leftImageView.image = image
         textField.leftView = leftImageView
         textField.leftViewMode = .always
     }
+
```
Comment (zuhair5) [inline ShopGo/CreateScreenViewController.swift:75]: remove space

File: ShopGo/CreateScreenViewController.swift [hunk 4]
```diff
@@ -39,10 +65,74 @@ class CreateScreenViewController: UIViewController {
...
+                nextButton.isEnabled = true
+                    nextButton.alpha = 1.0
+            }
+            else
+            {
+                nextButton.isEnabled = false
+                nextButton.alpha = 0.5
+            }
+        }
 }
```
Comment (zuhair5) [inline ShopGo/CreateScreenViewController.swift:130]: fix indentation

File: ShopGo/CreateAccountExtension.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
+//
+//  CreateAccountExtension.swift
+//  ShopGo
+//
+//  Created by Anoosha Wahid  on 28/09/2022.
+//
+
+import Foundation
+import UIKit
+
```
Comment (zuhair5) [inline ShopGo/CreateAccountExtension.swift:10]: remove space

File: ShopGo/CreateScreenViewController.swift [hunk 6]
```diff
@@ -7,7 +7,7 @@
 
 import UIKit
```
[the remark this answers is not in the document]
Comment (AnoshaKhan) [reply ShopGo/CreateScreenViewController.swift:8]: removed

File: ShopGo/CreateScreenViewController.swift (hunk 1 above)
Comment (AnoshaKhan) [reply ShopGo/CreateScreenViewController.swift]: removed

File: ShopGo/CreateScreenViewController.swift (hunk 4 above)
Comment (AnoshaKhan) [reply ShopGo/CreateScreenViewController.swift]: fixed

**Review by zuhair5 — COMMENTED** [review]
The story is good to be merged

**Review by zuhair5 — APPROVED** [review]

**Review by haroonmaqsoodch — APPROVED** [review]

**Pull request merged**